mcodingbot/tasks/stats.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+import traceback
+from dataclasses import dataclass
+from math import log
+from typing import TYPE_CHECKING, Any
+
+from mcodingbot.config import CONFIG
+
+if TYPE_CHECKING:
+    from mcodingbot.bot import Bot
+
+
+LOGGER = logging.getLogger(__file__)
+
+
+async def loop_update_channels(bot: Bot) -> None:
+    while True:
+        try:
+            await update_channels(bot)
+        except Exception:
+            LOGGER.error("Failed to update channel stats:", exc_info=True)
+
+        await asyncio.sleep(5 * 60)
+
+
+async def update_channels(bot: Bot) -> None:
+    stats = await get_stats(bot)
+
+    sub_channel = bot.cache.get_guild_channel(CONFIG.sub_count_channel)
+    view_channel = bot.cache.get_guild_channel(CONFIG.view_count_channel)
+    member_channel = bot.cache.get_guild_channel(CONFIG.member_count_channel)
+
+    assert sub_channel
+    assert view_channel
+    assert member_channel
+
+    await sub_channel.edit(name=f"Subs: {display_stats(stats.subs)}")
+    await view_channel.edit(name=f"Views: {display_stats(stats.views)}")
+
+    guild = bot.cache.get_guild(CONFIG.mcoding_server)
+    assert guild
+    member_count = guild.member_count
+    assert member_count is not None
+
+    await member_channel.edit(name=f"Members: {display_stats(member_count)}")
+
+
+@dataclass
+class Stats:
+    subs: float
+    views: float
+
+
+BASE_URL = "https://www.googleapis.com/youtube/v3/channels"
+_last_known_stats = Stats(0, 0)
+
+
+async def get_stats(bot: Bot) -> Stats:
+    link = (
+        f"{BASE_URL}?part=statistics&id={CONFIG.mcoding_yt_id}"
+        f"&key={CONFIG.yt_api_key}"
+    )
+
+    session = await bot.get_session()
+    async with session.get(link) as res:
+        res.raise_for_status()
+        response: dict[str, Any] = await res.json()
+
+    if not response:
+        LOGGER.error("Received 2XX but no data.")
+        return _last_known_stats
+
+    items = response.get("items")
+    if not items:
+        LOGGER.error("Response did not contain 'items'.")
+        return _last_known_stats
+
+    channel = items[0]
+    if channel.get("id") != CONFIG.mcoding_yt_id:
+        LOGGER.error("Channel ID was not mCoding.")
+        return _last_known_stats
+
+    statistics = channel.get("statistics")
+    if not statistics:
+        LOGGER.error("mCoding channel did not contain 'statistics'.")
+        return _last_known_stats
```

Comment:
I don't like the repeated return here but atm I have no idea how to improve this

Reply:
Feels overkill. If we had like 10 if-else it would be worth it, but for just 3 I think this is fine.